File: Problem3_LongestPalindrome.cs
// Time Complexity :O(n)
// Space Complexity : O(1)
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this :

//Approach: While iterating through each char of the input string, 
//Store the char encountered in hashset until its pair is found. 
//If the pair is found,remove the char from hashset and increment the counter by 2.
//By the end of the iteration if the hashset is not empty, increment the counter by 1.

public class Solution {
    public int LongestPalindrome(string s) {
        if(s == null) return 0;
        int maxLength = 0;
        HashSet<char> set = new HashSet<char>();
        for(int i = 0; i < s.Length; i++ ) {
            char ch = s[i];
            if(set.Contains(ch)) {
                maxLength += 2;
                set.Remove(ch);
            }
            else {
                set.Add(ch);
            }
        }
        if(set.Count > 0) {
            maxLength += 1;
        }

        return maxLength;
    }
}

File: Problem3_LongestPalindrome.java
import java.util.HashSet;

//Use a Hashset to store the chars in the String S 
//Remove the character from the set once a pair of the same char is found
//For each pair of the same char found increase the count by 2
//At the end the set can be empty the return count
//or the set can contain the characters with odd number occurances in the string, then return count by incrementing with 1

// Time Complexity : O(n)
// Space Complexity : O(1)
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this : Yes

class Problem3_LongestPalindrome {
    public int longestPalindrome(String s) {
        int count=0;
        if(s==null || s.length()==1) return s.length();
        HashSet<Character> set = new HashSet<>();
        for(int i=0; i<s.length(); i++){
            char c=s.charAt(i);
            if(set.contains(c)){
                count+=2;
                set.remove(c);
            }else{
                set.add(c);
            }
        }
        return !set.isEmpty()? ++count:count;
    }
}